src/aioswitcher/api/messages.py

```diff
@@ -130,16 +131,20 @@ def get_thermostat_remote_id(self) -> str:
         remote_hex = unhexlify(self._hex_response)
         return remote_hex[84:92].decode().rstrip("\x00")
 
-    def get_shutter_position(self) -> int:
-        """Return the current shutter position."""
-        hex_pos = self._hex_response[152:154].decode()
+    def get_shutter_position(self, index: int) -> int:
+        """Return the current position of the shutter 0 <= pos <= 100."""
+        start_index = 152 + (index * 32 if index else 0)
+        end_index = start_index + 2
+        hex_pos = self._hex_response[start_index:end_index].decode()
         return int(hex_pos, 16)
 
-    def get_shutter_direction(self) -> ShutterDirection:
-        """Return the current shutter direction."""
-        hex_dir = self._hex_response[156:160].decode()
-        directions = dict(map(lambda s: (s.value, s), ShutterDirection))
-        return directions[hex_dir]
+    def get_shutter_direction(self, index: int) -> ShutterDirection:
+        """Return the current direction of the shutter (UP/DOWN/STOP)."""
+        start_index = 156 + (index * 32 if index else 0)
```

Comment:
The `if index` has no effect
```suggestion
        start_index = 156 + (index * 32)
```